app/elements/bar-zoomable-chart/bar-zoomable-chart.html

```diff
@@ -0,0 +1,316 @@
+<link rel='import' href='../../bower_components/polymer/polymer.html'>
+<script src='../../bower_components/d3/d3.min.js' charset='utf-8'></script>
+<dom-module id='bar-zoomable-chart'>
+    <style>
+        :host {
+            display: block;            
+        }
+        #barChart {
+            width:100%;
+            height: 400px;
+        }
+        
+        .axis {
+          font-size: 11px;
+          fill: var(--bar-zoomable-chart-axis-color, --secondary-text-color);
+        }
+
+        .overview {
+            margin-top: 300px;
+        }
+
+        .axis line, .axis path {
+          fill: none;
+          stroke: rgba(0,0,0,var(--dark-divider-opacity));
+          shape-rendering: crispEdges;
+        }   
+        .brush .extent {
+          stroke: #fff;
+          fill-opacity: .125;
+          shape-rendering: crispEdges;
+        }
+    </style>
+    <template>
+    <div class='layout horizontal'>
+      <div id='chartContainer' class='flex auto'>
+        <svg id='barChart'></svg>
+      </div>
+    </div>
+
+    </template>
+</dom-module>
+<script>
+  (function() {
+    'use strict';
+    /* globals d3 */
+    Polymer({
+      is: 'bar-zoomable-chart',
+      properties: {
+        data: {
+          type: Object,
+          value: function() {
+            return {};
+          },
+          notify: true
+        },
+        minX: {
+          type: String,
+          value: 'N/A',
+          notify: true
+        },
+        maxX: {
+          type: String,
+          value: 'N/A',
+          notify: true
+        }
+      },
+      observers: [
+        'checkData(data)'
+      ],
+      checkData: function() {
+        if(this.data) {
+          this.processData();
+          this.minX = this.getItemKey(this.data.data[0]);
+          this.maxX = this.getItemKey(this.data.data[this.data.data.length - 1]);
+          this.createElements();
+        } else {
+          this.minX = 'N/A';
+          this.maxX = 'N/A';
+        }
+      },
+      getItemKey: function(item) {
+        var format = d3.time.format('%Y-%m-%d');
+        var formattedDate = format(new Date(item.date));
+        return formattedDate;
+        // Just do this if using a time scale
+        // return new Date(item[this.dataKey]);
+      },
+      processData: function() {
+        var parseDate = d3.time.format('%d/%m/%Y');
+
+        if (this.data.data) {
+          this.data.data.map(function(record) {
+            record.date = parseDate(new Date(record.date)); // turn the date string into a date object
```

Comment:
This component shouldn't modify the input data because it would cause issues if you're using the same data in other components/pages.  Instead, add a new property to the component (something like "barChartData") and store the processed input data in that property.